krm-functions/nf-deploy-fn/design.md

```diff
@@ -0,0 +1,229 @@
+NF Deploy KRM function design
+========================================
+
+Purpose of the function
+-----------------------
+
+The `NF Deploy` KRM function is designed to be a part of the pipeline of a Nephio NF kpt package.
+
+Its primary purpose is to scan the entire package for `IPAllocation`, `VLANAllocation` and `NetworkAttachmentDefinition` upstream custom resources in the package and then generate it to a set of more specific (child) KRM resource, i.e. `UPFDeployment` resources. 
```

Comment:
The design might have changed based on Wim's latest SDK proposal . ALso I think it looks at DNN?